internal/datastore/postgres/postgres.go

```diff
@@ -375,6 +378,98 @@ func (pgd *pgDatastore) ReadWriteTx(
 	return datastore.NoRevision, err
 }
 
+const repairTransactionIDsOperation = "transaction-ids"
+
+func (pgd *pgDatastore) Repair(ctx context.Context, operationName string, outputProgress bool) error {
+	switch operationName {
+	case repairTransactionIDsOperation:
+		return pgd.repairTransactionIDs(ctx, outputProgress)
+
+	default:
+		return fmt.Errorf("unknown operation")
+	}
+}
+
+const batchSize = 10000
+
+func (pgd *pgDatastore) repairTransactionIDs(ctx context.Context, outputProgress bool) error {
+	conn, err := pgx.Connect(ctx, pgd.dburl)
+	if err != nil {
+		return err
+	}
+	defer conn.Close(ctx)
+
+	// Get the current transaction ID.
+	currentMaximumID := 0
+	if err := conn.QueryRow(ctx, queryCurrentTransactionID).Scan(&currentMaximumID); err != nil {
+		if !errors.Is(err, pgx.ErrNoRows) {
+			return err
+		}
+	}
+
+	// Find the maximum transaction ID referenced in the transactions table.
+	referencedMaximumID := 0
+	if err := conn.QueryRow(ctx, queryLatestXID).Scan(&referencedMaximumID); err != nil {
+		if !errors.Is(err, pgx.ErrNoRows) {
+			return err
+		}
+	}
+
+	// The delta is what this needs to fill in.
+	log.Ctx(ctx).Info().Int64("current-maximum", int64(currentMaximumID)).Int64("referenced-maximum", int64(referencedMaximumID)).Msg("found transactions")
+	counterDelta := referencedMaximumID - currentMaximumID
+	if counterDelta < 0 {
+		return nil
+	}
+
+	var bar *progressbar.ProgressBar
+	if isatty.IsTerminal(os.Stderr.Fd()) && outputProgress {
+		bar = progressbar.Default(int64(counterDelta), "updating transactions counter")
+	}
+
+	for i := 0; i < counterDelta; i++ {
+		var batch pgx.Batch
```

Comment:
Have we determined how much impact does this loop have over a production postgres? I think we should throttle it.

Reply:
It shouldn't matter; it should only be run while SpiceDB is offline

Reply:
Have that been made clear anywhere in the command description?